### ISSUE TITLE
com.hazelcast.jet.sql.SqlMetadataInJobConfigTest.test_dmlMetadata

### ISSUE THREAD

**Issue by olukas** [hazelcast/hazelcast#22215]
_master_ (commit e5a25c4b67ddd1cc44504eb3f7921d57476c292e)

Failed on MacOS with ZuluJDK8: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-MacOS-ZuluJDK8/112/testReport/com.hazelcast.jet.sql/SqlMetadataInJobConfigTest/test_dmlMetadata/

<details><summary>Stacktrace:</summary>

```
java.lang.AssertionError: expected:<1> but was:<0>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:647)
	at org.junit.Assert.assertEquals(Assert.java:633)
	at com.hazelcast.jet.sql.SqlMetadataInJobConfigTest.test_dmlMetadata(SqlMetadataInJobConfigTest.java:118)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

</details>

Standard output can be found here - https://console.aws.amazon.com/s3/buckets/j-artifacts/Hazelcast-master-MacOS-ZuluJDK8/112/

**Comment by olukas** [comment]
Failed on Windows with OracleJDK8: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-Windows-OracleJDK8/414/testReport/com.hazelcast.jet.sql/SqlMetadataInJobConfigTest/test_dmlMetadata/

**Comment by olukas** [comment]
Failed on MacOS with ZuluJDK8: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-MacOS-ZuluJDK8/121/testReport/com.hazelcast.jet.sql/SqlMetadataInJobConfigTest/test_dmlMetadata/

**Comment by viliam-durina** [comment]
Failed on PR builder: https://jenkins.hazelcast.com/job/Hazelcast-pr-builder/13624/testReport/junit/com.hazelcast.jet.sql/SqlMetadataInJobConfigTest/test_dmlMetadata/

**Comment by krzysztofslusarski** [comment]
I see a flow that should be fixed by #22448. Closing for now.

**Comment by Patras3** [comment]
Failed again on _5.2.z_ with OracleJDK8: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-5.maintenance-OracleJDK8/55/testReport/junit/com.hazelcast.jet.sql/SqlMetadataInJobConfigTest/test_dmlMetadata/

Waiting for merge of mentioned PR.
